src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        * `MeshVolumeRegion`
+        * `PolygonalFootprintRegion`
+        """
+        # If one of the regions isn't fixed fall back on default behavior
+        if needsSampling(self) or needsSampling(other):
+            return super().difference(other)
+
+        # If other region is represented by a mesh, we can extract the mesh to
+        # perform boolean operations on it
+        if isinstance(other, MeshVolumeRegion):
+            other_mesh = other.mesh
+
+            # Compute difference using Trimesh (CalledProcessError usually means empty intersection for OpenSCAD)
+            try:
+                new_mesh = self.mesh.difference(other_mesh, engine=self.engine, debug=debug)
+            except CalledProcessError:
+                if self.engine == "scad":
+                    return EmptyRegion("EmptyMesh")
+                else:
+                    raise RuntimeError("Mesh boolean operation failed.")
+            except ValueError as exc:
+                raise ValueError("Unable to compute mesh boolean operation. Do you have the Blender and OpenSCAD installed on your system?") from exc
+
+            if new_mesh.is_empty:
+                return EmptyRegion("EmptyMesh")
+            elif new_mesh.is_volume:
+                return MeshVolumeRegion(new_mesh, tolerance=min(self.tolerance, other.tolerance), centerMesh=False, engine=self.engine)
+            else:
+                # Something went wrong, abort
+                return super().difference(other)
+
+        if isinstance(other, PolygonalFootprintRegion):
+            # Other region is a polygonal footprint region. We can bound it in the vertical dimension
+            # and then calculate the difference with the resulting mesh volume.
+
+            # Determine the mesh's vertical bounds (adding a little extra to avoid mesh errors) and
+            # the mesh's vertical center.
+            vertical_bounds = (self.mesh.bounds[0][2], self.mesh.bounds[1][2])
+            mesh_height = vertical_bounds[1] - vertical_bounds[0] + 1
+            centerZ = (vertical_bounds[1] + vertical_bounds[0])/2
+
+            # Compute the bounded footprint and recursively compute the intersection
+            bounded_footprint = other.approxBoundFootprint(centerZ, mesh_height)
+
+            return self.difference(bounded_footprint)
+
+        # Don't know how to compute this difference, fall back to default behavior.
+        return super().difference(other)
+
+
+    def uniformPointInner(self):
+        # TODO: Look into tetrahedralization, perhaps to be turned on when a heuristic
+        # is met. Currently using Trimesh's rejection sampling.
+        sample = trimesh.sample.volume_mesh(self.mesh, self.num_samples)
+
+        if len(sample) == 0:
+            raise RejectionException("Rejection sampling MeshVolumeRegion failed.")
+        else:
+            return Vector(*sample[0])
+
+    def distanceTo(self, point):
```

Comment:
Better make this a `distributionFunction` since the user can pass random points into it through the `distance to` operator.